File: Perl/stardate.pl
#!/usr/bin/perl


if ($#ARGV >= 0) {
	$now = `date -d '$ARGV[0]'`;
} else {
	$now = `date --utc`;
}
chomp($now);
$yp = 1 + ($y = 0 + `date +"%Y" -d '$now'`);

$y0 = `date -d '$y-01-01 00:00:00 UTC' +'%s'`;
$y1 = `date -d '$yp-01-01 00:00:00 UTC' +'%s'`;

$s=`date -d '$now' +'%s'`;
print $y+($s-$y0)/($y1-$y0)."\n";

File: Dart/stardate.dart
#!/usr/bin/env dart

double StarDate(d) {
  var y = d.toUtc().year;
  var t0 = new DateTime.utc(y).millisecondsSinceEpoch;
  var t1 = new DateTime.utc(y + 1).millisecondsSinceEpoch;
  return y + (d.millisecondsSinceEpoch - t0).toDouble()/(t1 - t0);
}

void main() {
  print(StarDate(new DateTime.now()));
}


